### DISCUSSION TITLE
SVG with text rendering and font loading

### DISCUSSION THREAD

**Discussion by ggerla** [pixijs/pixijs#7448] [Q&A]
Hi 
I have an issue to render text into svg with the right font.
For example suppose to have this svg

<svg xmlns="http://www.w3.org/2000/svg" height="150" width="1000"><path d="M1.238 46.5c0-1.381 2.744-2.5 6.128-2.5h985.22c3.384 0 6.127 1.119 6.127 2.5v46c0 1.381-2.742 2.5-6.127 2.5H7.364c-3.382 0-6.126-1.119-6.126-2.5z" fill="#393838"/><path d="M452.53 79.504c1.54 0 2.5-1.67 1.73-3l-7.53-13.01c-.77-1.33-2.69-1.33-3.46 0l-7.53 13.01c-.77 1.33.19 3 1.73 3zm-7.53-7c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1zm-1 2v2h2v-2z" fill="none"/><text style="line-height:1.25" x="27" y="80" font-weight="400" font-size="22" font-family="Roboto" letter-spacing="0" word-spacing="0" fill="#fff">ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz</text></svg>

and the following code:

let sprite = new PIXI.Sprite();
let texture = PIXI.Texture.from('data:image/svg+xml;charset=utf8,' + encodeURIComponent(svgContent));
sprite .texture = texture;
sprite.position.x = 0;
sprite.position.y = 0;

When the sprite is added to the stage and it is rendered the font used is NOT Roboto.
I spent a lot of time to understand the reason and finally I found this solution 

https://stackoverflow.com/questions/42402584/how-to-use-google-fonts-in-canvas-when-drawing-dom-objects-in-svg

Finally, I changed the svg manually adding the section <defs><style>...</style></defs> with the entire font definition (I don't add the cose because it is too long) and the font rendered is correct.

Is there a way to load the font and append it to the SVG at runtime into PIXI code?
Is there an alternative way to solve this problem?

**Comment by eXponenta** [comment]
Before load a SVG with specific font you MUST load font. 
Fonts is lazy-loaded, and fonts begin fetching only when canvas render SVG (yep, we render SVG to canvas via `drawImage`).
There are not events or something like events for lazy-loaded fonts.
Pixi can't handle this behaviour automatically.

**Comment by ggerla** [comment]
I tryed to "pre-load" the font but it doesn't work.
After the preload if I use the fillText on the context the font is corrent.
But the drawImage use a different font.

Do you have some working example?

**Comment by eXponenta** [comment]
If `drawImage` return another font - there are not anothere way except use shape-pased a SVG loader.
But there are not garanties that it will load correctlly.

https://github.com/SukantPal/pixi-essentials/tree/master/packages/svg

Easy solution - convert font-bassed text to path, all vector editors can do this (Illustrator, Inkscape , Figma)

**Comment by bigtimebuddy** [comment]
Are you using foreignObject in the SVG? The only way to
Do this is base64 encode the font like that stackoverflow thread. Sorry this is an SVG limitation not a pixi issue. I bumped into it while making https://github.com/pixijs/html-text

**Comment by ggerla** [comment]
> If drawImage return another font - there are not anothere way except use shape-pased a SVG loader.
> But there are not garanties that it will load correctlly.
> https://github.com/SukantPal/pixi-essentials/tree/master/packages/svg

if I well remember this solution has a better quality but it is slower.

> Easy solution - convert font-bassed text to path, all vector editors can do this (Illustrator, Inkscape , Figma)

I will try this solutions, but I'm not sure this is applicable for my use case.

thanks

**Comment by ggerla** [comment]
> Are you using foreignObject in the SVG? The only way to
> Do this is base64 encode the font like that stackoverflow thread. Sorry this is an SVG limitation not a pixi issue. I bumped into it while making https://github.com/pixijs/html-text

It is not an SVG limitation. It is  limitation of Image API from HTML5. In any case I agree that it is not a pixi issue.
I was asking if it is possible to have an enhancement in PIXI, not a fix. Probably I put my question in the wrong place. Sorry.
